prefabs/player.tscn

```diff
@@ -306,43 +369,93 @@ Vector2i(1, 1): {
+[node name="AudioListener2D" type="AudioListener2D" parent="PlayerController/Player Audio Manager"]
 
-[node name="Landing SFX" type="AudioStreamPlayer2D" parent="Player Audio Manager"]
-position = Vector2(584, 373)
+[node name="Landing SFX" type="AudioStreamPlayer2D" parent="PlayerController/Player Audio Manager"]
 stream = SubResource("AudioStreamInteractive_r8y8h")
 
-[node name="Jumping SFX" type="AudioStreamPlayer2D" parent="Player Audio Manager"]
-position = Vector2(584, 373)
+[node name="Jumping SFX" type="AudioStreamPlayer2D" parent="PlayerController/Player Audio Manager"]
 stream = SubResource("AudioStreamInteractive_r8y8h")
 
-[node name="Footstep SFX" type="AudioStreamPlayer2D" parent="Player Audio Manager"]
-position = Vector2(584, 373)
+[node name="Footstep SFX" type="AudioStreamPlayer2D" parent="PlayerController/Player Audio Manager"]
 stream = SubResource("AudioStreamInteractive_j32pn")
 
-[node name="Moth Jump SFX" type="AudioStreamPlayer2D" parent="Player Audio Manager"]
-position = Vector2(584, 373)
+[node name="Moth Jump SFX" type="AudioStreamPlayer2D" parent="PlayerController/Player Audio Manager"]
 stream = SubResource("AudioStreamInteractive_v6fwb")
 bus = &"SFX"
+
+[node name="Marker2D" type="Marker2D" parent="PlayerController"]
+position = Vector2(24, -18)
+
+[node name="Area2D" type="Area2D" parent="PlayerController"]
+position = Vector2(24, -18)
+scale = Vector2(2, 2)
+collision_mask = 2
+
+[node name="CollisionPolygon2D" type="CollisionPolygon2D" parent="PlayerController/Area2D"]
+polygon = PackedVector2Array(0, 0, 120, 90, 120, -90)
+
+[node name="LightComponent" type="PointLight2D" parent="PlayerController"]
+shadow_enabled = true
+texture = ExtResource("28_5627b")
+script = ExtResource("29_glieu")
+
+[node name="Camera2D" type="Camera2D" parent="PlayerController"]
+zoom = Vector2(2, 2)
+position_smoothing_enabled = true
+
+[node name="BaseHealthAndStatusComponent" parent="." instance=ExtResource("28_50nj0")]
+
+[node name="PlayerInputComponent" parent="." instance=ExtResource("29_tkue1")]
+
+[node name="RangedAttack" parent="." node_paths=PackedStringArray("projectile_spawn_position") instance=ExtResource("30_5ht77")]
+projectile_spawn_position = NodePath("../PlayerController/Marker2D")
+
+[node name="PlayerAttackComponent" type="Node" parent="." node_paths=PackedStringArray("lock_on_area", "attacks", "measurement_position", "projectile_spawn_position")]
+script = ExtResource("31_a0bar")
+lock_on_area = NodePath("../PlayerController/Area2D")
+attacks = [NodePath("../RangedAttack")]
+measurement_position = NodePath("../PlayerController/Marker2D")
+projectile_spawn_position = NodePath("../PlayerController/Marker2D")
+
+[connection signal="on_horizontal_movement" from="PlayerController" to="PlayerController/AnimatedSprite2D/AnimationPlayer" method="walking"]
+[connection signal="on_horizontal_movement_cancelled" from="PlayerController" to="PlayerController/AnimatedSprite2D/AnimationPlayer" method="idle"]
```

Comment:
I wouldnt hook up signals in group environment. It makes it hard to find connections when there is no way to reference them.

Reply:
What do you mean by this? This was from the player.tscn, this is just how I referenced the signals in the GUI

Reply:
The entity class was made specifically to deal with signals. 
In projects with more than one person it gets hard to mange signals in editor. I have done it in the past with Unity and if the person isnt available to tell you where it is then you cannot search it easily like you can in code